# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1460593627 -36000
#      Thu Apr 14 10:27:07 2016 +1000
# Node ID 555b6c78d6775f532c9e329dc9229a86da1b3e81
# Parent  fed1111e18998c34546a4ebecf88be039b754a17
Fix regression preventing "dbstop <line>" in debug mode (bug #47621)

* debug.cc (parse_dbfunction_params): Get cannonical name for function
instead of file name.

diff --git a/libinterp/corefcn/debug.cc b/libinterp/corefcn/debug.cc
--- a/libinterp/corefcn/debug.cc
+++ b/libinterp/corefcn/debug.cc
@@ -285,18 +285,17 @@ parse_dbfunction_params (const char *who
             else if (seen_if)
                 error ("%s: line number must come before 'if' clause\n");
             seen_at = true;
 
             if (! seen_in)
               {
                 // It was a line number. Get function name from debugger.
                 if (Vdebugging)
-                  //symbol_name = get_user_code ()->name ();
-                  symbol_name = get_user_code ()->fcn_file_name ();
+                  symbol_name = get_user_code ()->profiler_name ();
                 else
                   error ("%s: function name must come before line number "
                          "and 'if'", who);
                 seen_in = true;
               }
             else if (seen_if)
               error ("%s: line number must come before 'if' clause\n");
 
